starter-code/js/index.js

```diff
@@ -1,52 +1,79 @@
-function deleteItem(e){
-
+function deleteItem(itemNode){
+  itemNode.parentNode.removeChild(itemNode);
+  updateAllPrices();
 }
 
 function getPriceByProduct(itemNode){
-
-}
-
-function updatePriceByProduct(productPrice, index){
-
-}
-
-function getTotalPrice() {
-
+  var price = itemNode.querySelector('.price>span').innerHTML || 0;
+  var qty = itemNode.querySelector('input').value || 0;
+  return qty*price;
 }
 
-function createQuantityInput(){
-
+function updatePriceByProduct(itemNode, productPrice){
+  itemNode.querySelector('.total-price-product>span').innerHTML = productPrice;
 }
 
-function createDeleteButton(){
+function updateAllPrices() {
+  var allItems = document.querySelectorAll('.product');
+  var totalPriceHTML = document.querySelector('.total-price>span');
+  var totalPrice = 0;
 
-}
-
-function createQuantityNode(){
+  //Calculate all prices
+  for(var i = 0; i < allItems.length; i++ ){
+    var price = getPriceByProduct(allItems[i]);
+    updatePriceByProduct(allItems[i],price);
+    totalPrice += price;
+  }
 
+  totalPriceHTML.innerHTML = totalPrice;
 }
 
-function createItemNode(dataType, itemData){
-
+function createNewItem(){
+  var createNameInput = document.getElementById('create-name');
+  var createPriceInput = document.getElementById('create-price');
+  var name = createNameInput.value || "Item ";
+  var price = createPriceInput.value || 1 ;
+  createNameInput.value="";
+  createPriceInput.value="";
+
+  var productBluePrint = '<div class = "product"><div class="name container"><span>'+name+
+  '</span></div><div class="price container"><span>'+price+
+  '</span></div><div class="qty container"><label>QTY</label><input class="qty-input" value="1" type="number" min="0" max="100"></input></div><div class="total-price-product container"><span>0</span></div><div class="delete btn btn-delete container">DELETE</div></div>';
+
+  // create div and prod list
+  var newItem = document.createElement('div');
+  newItem.innerHTML = productBluePrint;
+
+  var productList = document.getElementById('product-list');
+  productList.appendChild(newItem);
+
+  updateAllPrices();
+  addEventsToDeleteButtons();
+  addEventsToQtyInputs();
 }
```

Comment:
Nice work.  You're page updates the price as soon as the user inputs the number.